### ISSUE TITLE
Crash at startup after update (1.2.3 → 1.3.1) during server certificate verification

### ISSUE THREAD

**Issue by omar-polo** [skyjake/lagrange#243]
Hi,

After the update from 1.2.3 to 1.3.1 I noticed that using the "old" profile lagrange crashes every time I try to load (or reload an existing) page.  With an empty one everything works perfectly fine.

I'm running OpenBSD 6.8 (soon to be 6.9) -CURRENT.

Here's the backtrace:

```
% egdb `which lagrange` lagrange.core
GNU gdb (GDB) 7.12.1
 <...snip...>
Core was generated by `lagrange'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  X509_STORE_get0_objects (xs=0x0) at /usr/src/lib/libcrypto/x509/x509_lu.c:758
758             return xs->objs;
[Current thread is 1 (process 494014)]
(gdb) bt
#0  X509_STORE_get0_objects (xs=0x0) at /usr/src/lib/libcrypto/x509/x509_lu.c:758
#1  0x00000aeb23b95a0b in X509_verify_cert (ctx=0xaea45993800) at /usr/src/lib/libcrypto/x509/x509_vfy.c:660
#2  0x00000ae83e76faea in verify_TlsCertificate (d=0xaea4598bcc0)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/lib/the_Foundation/src/tlsrequest.c:349
#3  0x00000ae83e6b734a in checkTrust_GmCerts (d=0xaea8f81db80, domain=..., cert=0xaea4598bcc0)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/src/gmcerts.c:414
#4  0x00000ae83e6c081c in checkServerCertificate_GmRequest_ (d=0xaea4ab2ae00)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/src/gmrequest.c:163
#5  0x00000ae83e6c0668 in processIncomingData_GmRequest_ (d=0xaea4ab2ae00, data=<optimized out>)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/src/gmrequest.c:219
#6  0x00000ae83e6bfcad in readIncoming_GmRequest_ (d=0xaea4ab2ae00, req=<optimized out>)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/src/gmrequest.c:240
#7  0x00000ae83e77116f in checkReadyRead_TlsRequest_ (d=0xaea4ab2a500)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/lib/the_Foundation/src/tlsrequest.c:724
#8  readIncoming_TlsRequest_ (d=0xaea4ab2a500)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/lib/the_Foundation/src/tlsrequest.c:744
#9  0x00000ae83e770d68 in run_TlsRequest_ (thread=<optimized out>)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/lib/the_Foundation/src/tlsrequest.c:755
#10 0x00000ae83e76710b in run_Threads_ (arg=0xaead784cc00)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/lib/the_Foundation/src/thread.c:84
#11 0x00000aea7938f2f1 in _rthread_start (v=<optimized out>) at /usr/src/lib/librthread/rthread.c:96
#12 0x00000aeab465e9fa in __tfork_thread () at /usr/src/lib/libc/arch/amd64/sys/tfork_thread.S:84
(gdb) f 2
#2  0x00000ae83e76faea in verify_TlsCertificate (d=0xaea4598bcc0)
    at /usr/ports/pobj/lagrange-1.3.1/lagrange-1.3.1/lib/the_Foundation/src/tlsrequest.c:349
349         const int result = X509_verify_cert(store);
(gdb) p store
$1 = (X509_STORE_CTX *) 0xaea45993800
(gdb) p *store
$2 = {ctx = 0x0, current_method = 0, cert = 0xaea45968700, untrusted = 0xaea4596eda0, crls = 0x0, param = 0xaea4599a600,
  other_ctx = 0x0, verify = 0xaeb23b97f30 <internal_verify>, verify_cb = 0xaeb23b981c0 <null_callback>,
  get_issuer = 0xaeb23aff4e0 <X509_STORE_CTX_get1_issuer>, check_issued = 0xaeb23b981f0 <check_issued>,
  check_revocation = 0xaeb23b96ad0 <check_revocation>, get_crl = 0x0, check_crl = 0xaeb23b98280 <check_crl>,
  cert_crl = 0xaeb23b985d0 <cert_crl>, check_policy = 0xaeb23b98690 <check_policy>,
  lookup_certs = 0xaeb23aff0c0 <X509_STORE_get1_certs>, lookup_crls = 0xaeb23aff2d0 <X509_STORE_get1_crls>, cleanup = 0x0,
  valid = 0, last_untrusted = 0, chain = 0x0, tree = 0x0, explicit_policy = 0, error_depth = 0, error = 17,
  current_cert = 0x0, current_issuer = 0x0, current_crl = 0x0, current_crl_score = 0, current_reasons = 0, parent = 0x0,
  ex_data = {sk = 0x0}}
(gdb)
```

Let me know if I can provide more information.

In the old profile (that I was using from ~1.0 up to now) I had some subscriptions, some client certs and some tabs opened: nothing really sensitive, so I can send you a tarball of the old ~/.config/lagrange directory if it helps troubleshooting this issue, but anyway nothing I'd like to upload here on github :)

Thanks,

**Comment by skyjake** [comment]
The new feature in v1.3 regarding this is the CA based certificate verification. I suppose it's trying to verify without the required OpenSSL objects in place.

This may fix it: 42691b66514e903396d708e13721b9d457be5d03. Could you try this on your old profile dir?

**Comment by omar-polo** [comment]
Yes, this fixes it, thanks!

**Comment by skyjake** [comment]
Fixed for v1.3.2 then.
